### PR TITLE
Add eslint rules to guard against unexpected imports/exports

### LINKED ISSUE #28510: Build process: __experimentalTruncate warning in `npm run dev`
Seeing a warning message output in my `npm run dev` terminal, this is despite a fresh `npm ci`, as well as a `rm -rf ./node_modules && npm install`:

```
[0] WARNING in ./packages/components/build-module/index.js 102:0-63
[0] "export 'default' (reexported as '__experimentalTruncate') was not found in './truncate'
```

It has been suggested this warning comes from a tiny typo in https://github.com/WordPress/gutenberg/pull/28176, and that we might enhance our linter to help catch such in the future.

### PR DESCRIPTION
**Pull request by kevin940726**
## Description

Fix #28510.

Add `eslint-plugin-import` and enable the `import/default` and `import/named` rules.

A typo is introduced in #28176, after briefly discussed on [Slack](https://wordpress.slack.com/archives/C02QB2JS7/p1611733941106900?thread_ts=1611733353.106200&cid=C02QB2JS7), I think it's best to update the linter to guard against these kinds of bugs.

This PR also fixes the errors found by the linter.

## How has this been tested?

Only apply the first commit of this PR and run the linter:

```sh
npm run lint-js
```

Observe that there're errors reported.

After applying all the changes, run the linter again, observe that there're no more errors.

## Types of changes

Bug fix

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by gziolo** [comment]
Great idea to add those rules. I see there are several violations to fix first, though.

If you want to add it to the ESLint plugin with WordPress configs, I would recommend starting with the warning level to not fail their CI checks after a minor update of dependencies. This is a common strategy that other ESLint plugins use to roll out more strict checks gradually. Even looking at Gutenberg changes necessary to satisfy linter - it's a breaking change. The approach we used in the past was marking new rules at warning level in the package and promote to an error in the Gutenberg repo.

**Comment by kevin940726** [comment]
@gziolo That's an excellent idea! Will try!

File: packages/eslint-plugin/CHANGELOG.md [hunk 1]
```diff
@@ -2,14 +2,15 @@
...
+-   Enabled `import/default` and `import/named` rules in the `recommended` ruleset. [#28513](https://github.com/WordPress/gutenberg/pull/28513)
+
 ## 8.0.1 (2021-01-28)
 
 ### Bug Fix
 
 -   Add missing `eslint-plugin-import` npm dependency ([#28545](https://github.com/WordPress/gutenberg/pull/28545)).
-
-## 8.0.0 (2021-01-21)
-
```
Comment (gziolo) [inline packages/eslint-plugin/CHANGELOG.md:12]: It looks like a merge issue. It should not get removed.

File: package.json [hunk 2]
```diff
@@ -157,6 +157,7 @@
 		"enzyme": "3.11.0",
 		"equivalent-key-map": "0.2.2",
 		"eslint-plugin-eslint-comments": "3.1.2",
+		"eslint-plugin-import": "2.22.1",
```
Comment (gziolo) [inline package.json:160]: We use it in the whole project again 👍

You probably should rebase this branch again, remove the changes from the lock file and run `npm i`. There should be only one line updated.

**Review by gziolo — COMMENTED** [review]
Everything looks great as it goes for the ESLint changes.

There is some usual dance left with rebases and the package lock file.

File: packages/eslint-plugin/CHANGELOG.md (hunk 1 above)
Comment (kevin940726) [reply packages/eslint-plugin/CHANGELOG.md:12]: Oops! Thanks!

File: package.json (hunk 2 above)
Comment (kevin940726) [reply package.json:160]: Makes sense! Will try!

**Review by gziolo — APPROVED** [review]
Nice work, thank you for wrangling it 💯

**Pull request merged**